presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -216,14 +233,25 @@ public void setupClient(OkHttpClient.Builder builder)
             }
 
             if (useSecureConnection) {
-                setupSsl(
-                        builder,
-                        SSL_KEY_STORE_PATH.getValue(properties),
-                        SSL_KEY_STORE_PASSWORD.getValue(properties),
-                        SSL_KEY_STORE_TYPE.getValue(properties),
-                        SSL_TRUST_STORE_PATH.getValue(properties),
-                        SSL_TRUST_STORE_PASSWORD.getValue(properties),
-                        SSL_TRUST_STORE_TYPE.getValue(properties));
+                SslVerificationMode sslVerificationMode = SSL_VERIFICATION.getValue(properties).orElse(SslVerificationMode.valueOf(getDefaults().get(SSL_VERIFICATION.getKey())));
+                if (sslVerificationMode.equals(FULL) || sslVerificationMode.equals(CA)) {
+                    setupSsl(
+                            builder,
+                            SSL_KEY_STORE_PATH.getValue(properties),
+                            SSL_KEY_STORE_PASSWORD.getValue(properties),
+                            SSL_KEY_STORE_TYPE.getValue(properties),
+                            SSL_TRUST_STORE_PATH.getValue(properties),
+                            SSL_TRUST_STORE_PASSWORD.getValue(properties),
+                            SSL_TRUST_STORE_TYPE.getValue(properties));
+                }
+
+                if (sslVerificationMode.equals(CA)) {
+                    builder.hostnameVerifier((hostname, session) -> true);
+                }
+
+                if (sslVerificationMode.equals(NONE)) {
+                    setupInsecureSsl(builder);
```

Comment:
it is unintuitive that this belongs to `if (useSecureConnection) {` block.

I think we need to change the code and remove `useSecureConnection` and instead use `sslVerificationMode`